core/base/discreteGradient/DiscreteGradient.cpp

```diff
@@ -164,21 +164,27 @@ bool DiscreteGradient::isCellCritical(const int cellDim,
   }
 
   if(cellDim == 0) {
-    return ((*gradient_)[0][cellId] == -1);
+    return ((*gradient_)[0][cellId] != GHOST_GRADIENT)
```

Comment:
if I understand correctly your code, `GHOST_GRADIENT` is equal to -2 and NULL_GRADIENT to -1.
so if the gradient for that cell is equal to -1, it cannot be equal to -2.
so that original line does not even need to be modified, right?

Reply:
(or you can simply change the -1 into NULL_GRADIENT)